web-api/src/persistence/postgres/cases/getEligibleCasesForTrialCity.ts

```diff
@@ -57,44 +60,46 @@ export const getEligibleCasesForTrialCity = async ({
   // use batchGet instead of queries with Promise.all?
   const casePromises = dbCases.map(async c => {
     const [privatePractitioners, irsPractitioners] = await Promise.all([
-      query({
-        ExpressionAttributeNames: {
-          '#pk': 'pk',
-          '#sk': 'sk',
-        },
-        ExpressionAttributeValues: {
-          ':pk': `case|${c.docketNumber}`,
-          ':skPrefix': 'privatePractitioner|',
-        },
-        KeyConditionExpression: '#pk = :pk AND begins_with(#sk, :skPrefix)',
+      getPrivatePractitionersOnCase({
+        docketNumber: c.docketNumber,
         applicationContext,
       }),
-      query({
-        ExpressionAttributeNames: {
-          '#pk': 'pk',
-          '#sk': 'sk',
-        },
-        ExpressionAttributeValues: {
-          ':pk': `case|${c.docketNumber}`,
-          ':skPrefix': 'irsPractitioner|',
-        },
-        KeyConditionExpression: '#pk = :pk AND begins_with(#sk, :skPrefix)',
+      getIrsPractitionersOnCase({
+        docketNumber: c.docketNumber,
         applicationContext,
       }),
     ]);
 
-    return purgeDynamoKeys({
+    const dynamoData = purgeDynamoKeys<
+      any,
+      {
+        irsPractitioners: IrsPractitioner[];
+        privatePractitioners: PrivatePractitioner[];
+      }
+    >({
       ...c,
       irsPractitioners,
       privatePractitioners,
     });
+    return { ...c, ...dynamoData };
   });
 
   const fullEligibleCases = await Promise.all(casePromises);
 
-  const casesForReturn = fullEligibleCases.map(c => {
-    return c ? transformNullToUndefined(RawEligibleCaseEntity(c)) : undefined;
-  });
+  const casesForReturn = fullEligibleCases
+    .filter(c => c)
+    .map(c => {
+      return {
+        ...fromKyselyCase(c),
+        isSealed: !!c.isSealed,
+        irsPractitioners: c.irsPractitioners,
+        privatePractitioners: c.privatePractitioners,
+        docketNumberWithSuffix: Case.getDocketNumberWithSuffix({
+          docketNumber: c.docketNumber,
+          docketNumberSuffix: c.docketNumberSuffix,
+        }),
+      };
+    });
```

Comment:
@TomElliottFlexion I'm not sure how you feel about the changes in this file, but it is more type-safe now with the new mapper.

Reply:
I should have said, "Please let me know how feel about the changes in this file!" So pretend I said that :)

Reply:
I'll take your lead! I am fairly agnostic when it comes to "entities" here; I can see the value of having a `RawEligibleCase` that uses `fromKyselyCase`, but a proliferation of single use entities has downsides too